fpa/fastapi/types.go

```diff
@@ -1,72 +1,51 @@
 package fastapi
 
 import (
-	"time"
+	"errors"
 
-	"github.com/google/uuid"
 	"github.com/ipfs/go-cid"
+	"github.com/textileio/fil-tools/fpa"
 )
 
 var (
-	EmptyID = ID("")
+	// ErrConfigNotFound returned when instance configuration doesn't exist
+	// in ConfigStore.
+	ErrConfigNotFound = errors.New("config not found")
+	// ErrCidConfigNotFound returned when cid configuration doesn't exist in
+	// ConfigStore.
+	ErrCidConfigNotFound = errors.New("cid config not found")
+	// ErrCidInfoNotFound returned if no storing state exists for a Cid.
+	ErrCidInfoNotFound = errors.New("the cid doesn't have any saved state")
 )
 
-type ID string
+// ConfigStore is a repository for all state of a FastAPI.
+type ConfigStore interface {
```

Comment:
👍 An additional selling point for the interface: some remote DB with backup capabilities.